src/engine/include/engine/Entity.hpp

```diff
@@ -16,19 +16,30 @@ namespace lei3d
         glm::vec3 scale;
     };
 
+    class EntityComponent;
+
     class Entity
     {
     public:
-        Model* model = nullptr;    
+        Model* m_Model = nullptr;    
```

Comment:
I was considering making the mesh a component to add to the entity.
A model is also possible but might consider associating mesh with model via addComponent (make template function?). The mesh will inherit the component class ofc.
Entities can also be light source, etc.